### ISSUE TITLE
The dialog blocks application exit.

### ISSUE THREAD

**Issue by 60rntogo** [samhocevar/portable-file-dialogs#37]
If I attempt to close my program while a dialog is open, then the dialog persists and the program is blocked. It exits only after I explicitly close the dialog. This happens even if I destroy the `pfd::open_dialog` object while handling the request to exit. Is there anything in the API to programmatically close dialogs?

**Comment by samhocevar** [comment]
There is no API for that, but I can understand how `dialog->abort()` could be useful. It seems a bit challenging to implement properly on Windows, though. I'll have a look at it.

**Comment by 60rntogo** [comment]
Yes, that feature would be nice to have. Thanks for looking into this and for maintaining the library. It's very useful.

**Comment by samhocevar** [comment]
I already implemented this for Linux and Mac OS X in the [`feature/kill`](https://github.com/samhocevar/portable-file-dialogs/tree/feature/kill) branch. I called the function `dialog::kill()` because both “cancel” and ”abort” already have meanings in the context of UI dialogs, so I wanted to avoid confusion.

**Comment by samhocevar** [comment]
I now have a proof of concept of `message::kill()` (not yet for other classes) on Windows in 94e4976ff80bf9c0d22269505dace0a3927f10ef. It is still a bit flaky but worked for me. If you have some time to test it I would appreciate it!

**Comment by 60rntogo** [comment]
That was fast!

I have just tried it on Linux (I may have a chance to test on Windows within the next few days as well) and `kill` appears to work as expected. Unfortunately, `open_dialog` segfaults when I select more than one file.

Two minor comments:
- `-Wpedantic` complains about an extra `;` on line `1560`,
- `operator<<` on line `787` is not `inline` (although I assume that this is just for testing purposes).

**Comment by samhocevar** [comment]
Okay, `dialog::kill()` is now implemented for all dialog types. Final commit is 361bce73fc3ccda8f9e37c098b6aa71f06746bae and it has been merged to `master`. I have reviewed and tested the feature for quite some time and it seems pretty stable.

I was unable to reproduce your `open_dialog` segfault. Could you maybe provide a stack trace, in a new issue if possible?
